l10n.py

```diff
@@ -1,45 +1,60 @@
 #!/usr/bin/env python3
-#
-# Localization with gettext is a pain on non-Unix systems. Use OSX-style strings files instead.
-#
+"""Localization with gettext is a pain on non-Unix systems. Use OSX-style strings files instead."""
 
-import codecs
-from collections import OrderedDict
+import builtins
+import locale
 import numbers
 import os
-from os.path import basename, dirname, exists, isfile, isdir, join, normpath
+import pathlib
 import re
 import sys
+import warnings
+from collections import OrderedDict
+from contextlib import suppress
+from os.path import basename, dirname, exists, isdir, isfile, join
 from sys import platform
-from traceback import print_exc
-import builtins
+from typing import TYPE_CHECKING, Dict, Iterable, Optional, Set, TextIO, Union, cast
+
+if TYPE_CHECKING:
+    def _(x: str) -> str: ...
 
-import locale
 try:
     locale.setlocale(locale.LC_ALL, '')
```

Comment:
It bothers me slightly that we're now doing this both here, and later (given the import is before) in EDMarketConnector.py's `__main__`.  But, both are doing it when they are for good reasons, so I guess I can live with it.  Maybe add a comment to each site so it's clear that changing it 'here' won't necessarily change overall behaviour.

Reply:
Yeah Im not sure if we want to remove it here or not. For now I added the suggested comment

Reply:
For completeness sake this PR also needs to add the equivalent comment in EDMarketConnector.py

Reply:
Done in both now